### PR TITLE
Lab8.1

### PR DESCRIPTION
**Pull request by miz14**
1. Реализована GameCommand
2. Добавлен адаптер IReceiver для непотокобезопасной очереди

### CODE REVIEW

**Comment by Raipo7** [comment]
## [Codecov](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
Patch coverage: **`100.00`**% and no project coverage change.
> Comparison is base [(`ef195ff`)](https://app.codecov.io/gh/Raipo7/TeamN2/commit/ef195ff09fb3502c29e4933d103d5b3a350007b5?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) 100.00% compared to head [(`0956571`)](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) 100.00%.

:exclamation: Your organization is not using the GitHub App Integration. As a result you may experience degraded service beginning May 15th. Please [install the Github App Integration](https://github.com/apps/codecov) for your organization. [Read more](https://about.codecov.io/blog/codecov-is-updating-its-github-integration/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

<details><summary>Additional details and impacted files</summary>


```diff
@@            Coverage Diff             @@
##              main       #33    +/-   ##
==========================================
  Coverage   100.00%   100.00%            
==========================================
  Files           14        28    +14     
  Lines          315       520   +205     
  Branches        32        38     +6     
==========================================
+ Hits           315       520   +205     
```


| [Impacted Files](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) | Coverage Δ | |
|---|---|---|
| [SpaceBattle.Lib/ActionCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL0FjdGlvbkNvbW1hbmQuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/GameComand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL0dhbWVDb21hbmQuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/QueueReceiveAdapter.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1F1ZXVlUmVjZWl2ZUFkYXB0ZXIuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/HardStopCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvSGFyZFN0b3BDb21tYW5kLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/ReceiverAdapter.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvUmVjZWl2ZXJBZGFwdGVyLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SendCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VuZENvbW1hbmQuY3M=) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SenderAdapter.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VuZGVyQWRhcHRlci5jcw==) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/ServerThread.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU2VydmVyVGhyZWFkLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/SoftStopCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvU29mdFN0b3BDb21tYW5kLmNz) | `100.00% <100.00%> (ø)` | |
| [SpaceBattle.Lib/Threads/UpdateBehaviourCommand.cs](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#diff-U3BhY2VCYXR0bGUuTGliL1RocmVhZHMvVXBkYXRlQmVoYXZpb3VyQ29tbWFuZC5jcw==) | `100.00% <100.00%> (ø)` | |

... and [4 files with indirect coverage changes](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None)


</details>

[:umbrella: View full report in Codecov by Sentry](https://app.codecov.io/gh/Raipo7/TeamN2/pull/33?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).   
:loudspeaker: Do you have feedback about the report comment? [Let us know in this issue](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

File: SpaceBattle.Lib/GameComand.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    private Stopwatch time;
+
+    public GameCommand(object scope)
+    {
+        this.scope = scope;
+        IoC.Resolve<Hwdtech.ICommand>("Scopes.Current.Set", scope);
+
+        this.receiver = IoC.Resolve<IReceiver>("Game.GetReceiver");
+        this.gameItems = IoC.Resolve<IDictionary<string, object>>("Game.GetItems");
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/GameComand.cs]: --

File: SpaceBattle.Lib/GameComand.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private int gameTick;
+
+    private Stopwatch time;
+
+    public GameCommand(object scope)
+    {
+        this.scope = scope;
+        IoC.Resolve<Hwdtech.ICommand>("Scopes.Current.Set", scope);
+
+        this.receiver = IoC.Resolve<IReceiver>("Game.GetReceiver");
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/GameComand.cs]: Inject

File: SpaceBattle.Lib/GameComand.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    private object scope;
+    private int gameTick;
+
+    private Stopwatch time;
+
+    public GameCommand(object scope)
+    {
+        this.scope = scope;
+        IoC.Resolve<Hwdtech.ICommand>("Scopes.Current.Set", scope);
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/GameComand.cs]: exec

File: SpaceBattle.Lib/GameComand.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        this.receiver = IoC.Resolve<IReceiver>("Game.GetReceiver");
+        this.gameItems = IoC.Resolve<IDictionary<string, object>>("Game.GetItems");
+        this.gameTick = IoC.Resolve<int>("Game.GetTick");
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/GameComand.cs]: -> Execute

**Review by zaryanezrya — CHANGES_REQUESTED** [review]

**Comment by miz14** [comment]
Принято на занятии

**Pull request merged**